### PR TITLE
Expose running compaction in coordinator for the monitor

### PR DESCRIPTION
**Pull request by dlmarion**

### CODE REVIEW

File: core/src/main/thrift/compaction-coordinator.thrift [hunk 1]
```diff
@@ -43,9 +43,9 @@ enum TCompactionState {
 struct TCompactionStatusUpdate {
   1:TCompactionState state
   2:string message
-  3:i64 bytesToBeCompacted
-  4:i64 bytesRead
-  5:i64 bytesWritten
+  3:i64 entriesToBeCompacted
+  4:i64 entriesRead
+  5:i64 entriesWritten
 }
```
Comment (milleruntime) [inline core/src/main/thrift/compaction-coordinator.thrift:49]: I missed why these were changed. I did a rebase on your branch and this broke some stuff on my branch. I think its OK just wondering. Are these values not bytes?

Comment (dlmarion) [reply core/src/main/thrift/compaction-coordinator.thrift:49]: Yes, I'm pretty sure it's not bytes and it *is* entries. The genesis for those metrics is [here](https://github.com/dlmarion/accumulo/blob/expose-external-compaction-progress-for-monitor/server/compactor/src/main/java/org/apache/accumulo/compactor/Compactor.java#L684). If you dig into that you get to:

FileCompactor.getRunningCompactions() -> new CompactionInfo(FileCompactor)

**Comment by milleruntime** [comment]
I was able to test this with the page I am working on in the Monitor and I think it is good to go. The only thing I noticed that by having the Monitor make calls to the Compaction Coordinator, it introduces a Thrift dependency to the Monitor:
<pre>
[INFO] --- maven-dependency-plugin:3.1.2:analyze-only (analyze) @ accumulo-monitor ---
[WARNING] Used undeclared dependencies found:
[WARNING]    org.apache.thrift:libthrift:jar:0.15.0:compile
</pre>

I am OK with this. I was actually surprised there wasn't one already. I guess all the calls that go through the Manager don't require a direct dependency.

**Comment by ctubbsii** [comment]
> I was able to test this with the page I am working on in the Monitor and I think it is good to go. The only thing I noticed that by having the Monitor make calls to the Compaction Coordinator, it introduces a Thrift dependency to the Monitor:
> 
> [INFO] --- maven-dependency-plugin:3.1.2:analyze-only (analyze) @ accumulo-monitor ---
> [WARNING] Used undeclared dependencies found:
> [WARNING]    org.apache.thrift:libthrift:jar:0.15.0:compile
> I am OK with this. I was actually surprised there wasn't one already. I guess all the calls that go through the Manager don't require a direct dependency.

I can't reproduce that, so I'm assuming you're describing changes not included in this PR. In general, it's best to keep our Thrift types from leaking too far from the RPC layer. The monitor shouldn't interact with Thrift at all. It should interact with other service APIs, and those service APIs should not leak Thrift types, if at all possible.

**Comment by milleruntime** [comment]
> I can't reproduce that, so I'm assuming you're describing changes not included in this PR. 

Yes. I am using `CompactionCoordinatorService.Client` to call the method `getRunningCompactions`. It returns multiple thrift types, TExternalCompactionList and TExternalCompaction.

> The monitor shouldn't interact with Thrift at all. It should interact with other service APIs, and those service APIs should not leak Thrift types, if at all possible.

I don't understand what you mean. I am also not sure how the Monitor doesn't already have this dependency. The Monitor currently uses thrift types like `ManagerMonitorInfo` and `ManagerClientService` a great deal.

**Comment by ctubbsii** [comment]
> > I can't reproduce that, so I'm assuming you're describing changes not included in this PR.
> 
> Yes. I am using `CompactionCoordinatorService.Client` to call the method `getRunningCompactions`. It returns multiple thrift types, TExternalCompactionList and TExternalCompaction.

It's hard to troubleshoot without seeing your code.

> > The monitor shouldn't interact with Thrift at all. It should interact with other service APIs, and those service APIs should not leak Thrift types, if at all possible.
> 
> I don't understand what you mean. I am also not sure how the Monitor doesn't already have this dependency. The Monitor currently uses thrift types like `ManagerMonitorInfo` and `ManagerClientService` a great deal.

I guess there are two kinds of "Thrift types". There's our autogenerated code, like those you mention (I still think they shouldn't leak into the monitor, but I guess they're not causing a problem right now) that are from our accumulo-core jar. And then there's types from the Thrift jar. maven-dependency-plugin is checking for compile-time dependencies, not runtime dependencies. So, it shouldn't complain unless the monitor references a type `org.apache.thrift.*`. For the examples you gave, we're never actually referencing any `org.apache.thrift.*` classes in the monitor, even if those classes themselves reference them (creating a runtime dependency).

**Comment by milleruntime** [comment]
So it turns out I was introducing the thrift dependency by adding these two imports:
<pre>
-import org.apache.thrift.TException;
-import org.apache.thrift.transport.TTransportException;
</pre>
So I think what we have in the Monitor is a bit of a hack to get around this. By catching Exception, like we do when we get the ManagerClient, I can avoid importing the thrift exceptions and it doesn't complain about the dependency.

**Review by milleruntime — APPROVED** [review]

**Comment by dlmarion** [comment]
I'll resolve the conflict and merge this morning

**Comment by milleruntime** [comment]
Sorry I forgot to mention one of the values missing from TExternalCompaction. The compaction kind is available on the coordinator but not set in the thrift object when calling getRunningCompactions(). I think it can be pulled from the job object.

File: core/src/main/thrift/compaction-coordinator.thrift [hunk 2]
```diff
@@ -50,11 +50,9 @@ struct TCompactionStatusUpdate {
...
   1:string queueName
-  2:data.TKeyExtent extent
-  3:list<tabletserver.InputFile> files
-  4:string outputFile
-  5:string compactor
-  6:map<i64,TCompactionStatusUpdate> updates
+  2:string compactor
+  3:map<i64,TCompactionStatusUpdate> updates
+  4:tabletserver.TExternalCompactionJob job
 }
```
Comment (milleruntime) [inline core/src/main/thrift/compaction-coordinator.thrift:56]: Including the job here works perfectly! Thanks

**Review by milleruntime — APPROVED** [review]

**Pull request merged**